### PR TITLE
 changes for controller runtime upgrade

### LINKED ISSUE #69: Upgrade controller-runtime and operator-sdk
### Description
sigs.k8s.io/controller-runtime lib and the operator-sdk that we are using are quite old.

### Importance
must-have

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anisha.kj <anisha.kj@dell.com>

### Change log description

Update controller run time to 0.5.2 and operator-sdk to v0.17.0
### Purpose of the change

Fixes #69

### What the code does

With controller run time upgrade, webhook validation mechanism changes. We have to create webhook svc, Also for validation a certificate needs to be installed.

### How to verify it

1. Installing bookkeeper operator and bookkeeper cluster is verified by manual installation and helm charts.
2. Upgrading bookkeeper operator and bookkeeper cluster is verified by manual installation and helm charts.

### CODE REVIEW

**Review by Prabhaker24 — APPROVED** [review]
Just add the "how to verify" section in the PR description other than that this PR LGTM.

File: doc/operator-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#### Upgrade via helm
+
+The upgrade to Operator 0.1.3 can be triggered using the following command
+```
+helm upgrade <operator release name> <location of 0.1.3 charts>  --set webhookCert.generate=false --set webhookCert.certName=<cert-name> --set webhookCert.secretName=<secret-name>
+```
+#### Upgrade manually
+
+To manually trigger the upgrade to Operator 0.1.3, run the script `operatorUpgrade.sh` under [tools](https://github.com/pravega/bookkeeper-operator/blob/master/tools) folder. This script patches the and creates necessary K8s artifacts, needed by 0.1.3 Operator, prior to triggering the upgrade by updating the image tag in Operator deployment.
```
Comment (SrishT) [inline doc/operator-upgrade.md:43]: This script patches the ?

**Review by SrishT — CHANGES_REQUESTED** [review]

File: tools/operatorUpgrade.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+sed -i "s/value:.*/value: $op_name "/ ./manifest_files/patch.yaml
+
+sed -i "/imagePullPolicy:.*/{n;s/name.*/name: $op_name/}" ./manifest_files/patch.yaml
+
+#updating the operator using patch file
+kubectl patch deployment $op_name --namespace ${namespace} --type merge --patch "$(cat ./manifest_files/patch.yaml)"
+
+}
+
+UpgradingToBookkeeperoperator $1 $2 $3
```
[the remark this answers is not in the document]
Comment (anishakj) [reply tools/operatorUpgrade.sh]: Done

File: tools/manifest_files/patch.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        env:
+        - name: WATCH_NAMESPACE
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: OPERATOR_NAME
+          value: bkop-bookkeeper-operator 
+        image: kjanisha/bookkeeper-operator:openapi
```
[the remark this answers is not in the document]
Comment (anishakj) [reply tools/manifest_files/patch.yaml]: Done

File: doc/operator-upgrade.md (hunk 1 above)
Comment (anishakj) [reply doc/operator-upgrade.md]: Done

**Review by SrishT — APPROVED** [review]
LGTM

**Pull request merged**